### PR TITLE
[localize] Port localize decorator to hybrid

### PR DESCRIPTION
**Pull request by justinfagnani**
Part of https://github.com/lit/lit/issues/4144

This makes `@localize()` work with standard decorators, adds standard decorator tests, and adds a test case to `@lit/localize-tools` to ensure that the transformer works.

This is needed to make https://github.com/lit/lit/pull/4156 pass.

### CODE REVIEW

File: packages/localize/src/internal/localized-decorator.ts [hunk 1]
```diff
@@ -36,28 +43,18 @@ import type {
...
-    finisher(clazz: typeof ReactiveElement) {
-      clazz.addInitializer(updateWhenLocaleChanges);
-    },
+export const localized: Localized =
+  (): LocalizedDecorator =>
+  (
+    clazz: typeof ReactiveElement,
+    _context?: ClassDecoratorContext<typeof ReactiveElement>
+  ) => {
+    (clazz as typeof ReactiveElement).addInitializer(updateWhenLocaleChanges);
```
Comment (rictic) [inline packages/localize/src/internal/localized-decorator.ts:52]: is this cast needed?

File: packages/localize/src/internal/localized-decorator.ts [hunk 2]
```diff
@@ -36,28 +43,18 @@ import type {
...
-  return {
-    kind,
-    elements,
-    finisher(clazz: typeof ReactiveElement) {
-      clazz.addInitializer(updateWhenLocaleChanges);
-    },
+export const localized: Localized =
+  (): LocalizedDecorator =>
+  (
+    clazz: typeof ReactiveElement,
```
Comment (rictic) [inline packages/localize/src/internal/localized-decorator.ts:49]: Should we use the `Interface` trick to strip off non-public fields?

File: packages/localize/src/internal/localized-decorator.ts (hunk 1 above)
Comment (rictic) [inline packages/localize/src/internal/localized-decorator.ts:52]: Nice that it gets simpler and smaller with this change!

**Review by rictic — APPROVED** [review]
Great!

Taking on trust that the testdata files are actually copies

Very weird that github doesn't see them as such, since they should have the exact same content

File: packages/localize/src/internal/localized-decorator.ts (hunk 1 above)
Comment (justinfagnani) [reply packages/localize/src/internal/localized-decorator.ts:52]: nope!

File: packages/localize/src/internal/localized-decorator.ts (hunk 2 above)
Comment (justinfagnani) [reply packages/localize/src/internal/localized-decorator.ts:49]: done

File: packages/localize/src/internal/localized-decorator.ts (hunk 1 above)
Comment (justinfagnani) [reply packages/localize/src/internal/localized-decorator.ts:52]: Yes!

**Pull request merged**